### ISSUE TITLE
Eleven value rules write into a string the parser paired across a quotation mark inside a comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#508]
### What minimal example or steps are needed to reproduce the bug?

`postcss-value-parser` knows nothing of a comment opened by a double slash and closes one opening `/*/` on its own star, so a quotation mark written in the text of either opens a string to it that runs to the next quotation mark of the value — and from there on every opening mark of the file is a closing one to the parser and the other way round. Eleven rules walk the parse of the value as the file spells it and pass over the one node that opens inside the comment, through `findCommentSpanHolding`; the string the file spells behind the comment reaches them as words of the value and a string never closed, and they report and rewrite the text inside its quotation marks. Each rule under `--fix`, in plain CSS:

```
a { b: 2PX /*/ " */ "2PX"; }          → a { b: 2px /*/ " */ "2px"; }          unit-case, "lower"
a { b: 1.50px /*/ " */ "1.50px"; }    → a { b: 1.5px /*/ " */ "1.5px"; }      number-no-trailing-zeros
a { b: #FFF /*/ " */ "#FFF"; }        → a { b: #fff /*/ " */ "#fff"; }        color-hex-case, "lower"
a { b: f( 1 ) /*/ " */ "f( 1 )"; }    → a { b: f(1) /*/ " */ "f(1)"; }        function-parentheses-space-inside, "never"
a { b: f(1,2) /*/ " */ "f(1,2)"; }    → a { b: f(1, 2) /*/ " */ "f(1, 2)"; }  function-comma-space-after, "always"
```

each with a second warning drawn on the string — at `1:23`, `1:28`, `1:23`, `1:27` and `1:28` — beside the one the code in front of the comment draws. The same happens with a comment opened by a double slash under `postcss-scss`: `a { b: 2PX // "⏎"2PX"; }` comes back as `2px // "⏎"2px"`. Under `postcss-less` that spelling is refused as an unclosed string, since its tokenizer pairs the mark in the comment the same way, and the `/*/` spelling behaves as in plain CSS.

The rules: `color-hex-case`, `function-comma-newline-after`, `function-comma-newline-before`, `function-comma-space-after`, `function-comma-space-before`, `function-max-empty-lines`, `function-parentheses-newline-inside`, `function-parentheses-space-inside`, `number-leading-zero`, `number-no-trailing-zeros` and `unit-case`. The sweep `scripts/sweeps/quote-in-comment.ts` of the branch closing #504 found them and measured them on the base as well: every one of the eleven rewrites the string in a value and in a call's arguments, behind `/*/` under all three syntaxes and behind `//` under `postcss-scss`, with either quotation mark. The three other rules of #378 parse a copy with the comments blanked out and are not touched: `aspect-ratio-notation`, `string-quotes` and, since #504, `named-grid-areas-alignment`.

The same cut hides a call. `a { b: g(2PX /*/ " */ "2PX"); }` under `function-parentheses-space-inside: "always"` draws no warning at all, where `g(2PX /** " */ "2PX")` draws two: the parser never closes `g(`, since the string opened inside the comment took its closing parenthesis. Both `function-parentheses-*-inside` rules go quiet that way under their `always` options, on 28 and 32 texts of the sweep.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

One warning, for the unit standing in front of the comment; the text inside the quotation marks left exactly as written, since a string is no dimension, no number, no colour and no call; and a call closed where the file closes it.

### What actually happened?

A second warning on the string, and the text inside it rewritten by `--fix`.

### What is the cause?

Each rule walks `valueParser(value)` over the value as the file spells it and asks `findCommentSpanHolding` of every node's opening. The answer is right for the node that opens inside the comment and says nothing of the nodes behind it, which the parser cut around a string that is no string of the file. `unit-case` blanks the comments only to find the interpolations, and walks the raw parse like the others.

### Do you have a proposal to fix the bug?

Parse a copy of the value with every comment blanked — `blankComments` over `syntax.commentSpans` — the way `aspect-ratio-notation`, `string-quotes` and `named-grid-areas-alignment` do, and write every fix at the position the copy gives it in the text the file spells; the copy is as long as the value, so its positions are the file's. The four comma rules share `functionCommaSpaceChecker`, so the move is one place for them.
